http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -509,30 +553,39 @@ final class LeoKubernetesServiceInterp[F[_]: Parallel](
       //
       // There are DB constraints to handle potential name collisions.
       uid = s"${RandomStringUtils.randomAlphabetic(1)}${RandomStringUtils.randomAlphanumeric(5)}".toLowerCase
-      namespaceName <- KubernetesName.withValidation(
-        s"${uid}-${galaxyConfig.namespaceNameSuffix}",
-        NamespaceName.apply
-      )
-      release <- KubernetesName.withValidation(
-        s"${uid}-${galaxyConfig.releaseNameSuffix}",
-        Release.apply
-      )
+      namespaceId = lastUsedApp.fold(
+        NamespaceId(-1)
+      )(app => app.namespaceId)
+      namespaceName <- lastUsedApp.fold(
+        KubernetesName.withValidation(
+          s"${uid}-${galaxyConfig.namespaceNameSuffix}",
+          NamespaceName.apply
+        )
+      )(app => app.namespaceName.asRight[Throwable])
+
+      chart = lastUsedApp.map(_.chart).getOrElse(Chart(galaxyConfig.chartName, galaxyConfig.chartVersion))
```

Comment:
This would mean if we update the `galaxykubeman` chart version in Leo conf, it would not take effect for users who have an existing disk. 

Am a little concerned about this -- e.g. what if we need to roll out a gkm bugfix that does not affect disks in any way? Feels like we should have a way to do that..

Reply:
yea...I asked Alex about versioning..he did mention if it's patch upgrade, it should be non-breaking in theory...we definitely need to think about this a bit more...I'm thinking we can table this for this PR, and get it finalized before UI is in prod

Reply:
OK, thanks for making that ticket.. yeah maybe a policy like "patch versions are OK, major versions require a new disk" would make sense. There are also UX considerations as ideally users would be able to opt in to upgrading their Galaxy.

Reply:
so upgrading galaxy itself doesn't necessarily require chart upgrade..created https://broadworkbench.atlassian.net/browse/IA-2557 to track